File: dart/lib/htmlsanitizer.dart
/// Clean HTML of constructs that can lead to Cross-Site Scripting (XSS).
///
/// A thin `dart:ffi` binding over the monorepo's one shared native sanitizer core
/// (`core/native/libhtmlsanitizer.so`, compiled from pure Aether). It contains
/// **no sanitizer logic**: every member marshals to an `aether_hs_embed_*`
/// call. One sanitizer core, one set of behaviours, N language surfaces.
///
/// ```dart
/// import 'package:htmlsanitizer/htmlsanitizer.dart';
///
/// final s = HtmlSanitizer();
/// print(s.sanitize('<div onclick="alert(1)">Hello</div>'));  // <div>Hello</div>
/// s.close();
/// ```
library;

export 'src/sanitizer.dart'
    show
        Attribute,
        AllowList,
        FilterUrlHandler,
        HtmlSanitizer,
        Node,
        NodeKind,
        PostProcessHandler,
        Reason,
        RemovingAttributeHandler,
        RemovingCommentHandler,
        RemovingStyleHandler,
        RemovingTagHandler,
        sanitize,
        sanitizeDocument;

File: ruby/lib/htmlsanitizer.rb
# frozen_string_literal: true

# htmlsanitizer — clean HTML of XSS vectors.
#
# A thin Fiddle binding over ONE shared native sanitizer core
# (core/native/libhtmlsanitizer.so, built from pure Aether). No sanitizer
# logic lives in this gem; it only marshals values across the C ABI.
#
#     require "htmlsanitizer"
#
#     s = HtmlSanitizer.new
#     s.sanitize('<div onclick="alert(1)">hi</div>')  # => "<div>hi</div>"
#     s.close

require_relative "htmlsanitizer/version"
require_relative "htmlsanitizer/native"
require_relative "htmlsanitizer/sanitizer"

module HtmlSanitizer
  class << self
    # HtmlSanitizer.new is a shorthand for HtmlSanitizer::Sanitizer.new; with a
    # block it closes the handle for you.
    def new(native_lib: nil, &blk)
      Sanitizer.open(native_lib: native_lib, &blk)
    end

    # One-shot convenience: sanitize a fragment with the secure defaults.
    def sanitize(html, base_url = "", native_lib: nil)
      new(native_lib: native_lib) { |s| s.sanitize(html, base_url) }
    end

    # One-shot convenience for a whole document.
    def sanitize_document(html, base_url = "", native_lib: nil)
      new(native_lib: native_lib) { |s| s.sanitize_document(html, base_url) }
    end

    # The sanitizer core's ABI revision.
    def abi_version
      new { |s| s.abi_version }
    end
  end
end
